### PR TITLE
CCIP-3920 Fee boosting plugin level specs

### PR DESCRIPTION
**Pull request by mateusz-sekara**
Basic fee boosting spec on a plugin level, one scanario covering
* boosting message fee for multiple hours
* changing execution cost in the meantime
* filtering out too expensive messages

### CODE REVIEW

**Review by makramkd — APPROVED** [review]

File: execute/exectypes/costly_messages.go [hunk 1]
```diff
@@ -206,6 +220,10 @@ func (n *StaticMessageExecCostUSD18Calculator) MessageExecCostUSD18(
 	return messageExecCosts, nil
 }
 
+func (n *StaticMessageExecCostUSD18Calculator) UpdateCosts(msgID cciptypes.Bytes32, cost plugintypes.USD18) {
+	n.costs[msgID] = cost
```
[the remark this answers is not in the document]
Comment (mateusz-sekara) [reply execute/exectypes/costly_messages.go:224]: It's just a temporary implementation. After providing proper one, this one will be used only in tests

File: execute/exectypes/costly_messages.go [hunk 2]
```diff
@@ -222,6 +240,20 @@ type CCIPMessageFeeUSD18Calculator struct {
...
+	lggr logger.Logger,
+	ccipReader readerpkg.CCIPReader,
+	relativeBoostPerWaitHour float64,
+	now func() time.Time,
+) *CCIPMessageFeeUSD18Calculator {
+	return &CCIPMessageFeeUSD18Calculator{
+		lggr:                     lggr,
+		ccipReader:               ccipReader,
+		relativeBoostPerWaitHour: relativeBoostPerWaitHour,
+		now:                      now,
```
[the remark this answers is not in the document]
Comment (mateusz-sekara) [reply execute/exectypes/costly_messages.go:253]: It's not changed in this PR, but yeah it returns the current time. I guess introduced for easier testing for the struct

File: execute/test_utils.go [hunk 3]
```diff
@@ -203,51 +220,36 @@ func (it *IntTest) Start() *testhelpers.OCR3Runner[[]byte] {
 	)
 	require.NoError(it.t, err)
 
+	var feeCalculator exectypes.MessageFeeE18USDCalculator
+	if it.feeCalculator != nil {
+		feeCalculator = it.feeCalculator
+	} else {
+		feeCalculator = &exectypes.ZeroMessageFeeUSD18Calculator{}
```
[the remark this answers is not in the document]
Comment (mateusz-sekara) [reply execute/test_utils.go:227]: Types don't match, so I needed to explicitly say we want the interface type here 
```go
	var feeCalculator exectypes.MessageFeeE18USDCalculator
```

[the remark this answers is not in the document]
Comment (mateusz-sekara) [reply execute/test_utils.go:227]: Wouldn't ctor be an overkill here? It's just a simple struct without any fields.

File: execute/exectypes/costly_messages.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mateusz-sekara) [reply execute/exectypes/costly_messages.go:239]: Clarified that in the comment

**Pull request merged**